### PR TITLE
Only specified function could be supported in correlated subquery

### PR DESCRIPTION
**Pull request by EmmyMiao87**

### CODE REVIEW

**Comment by kangkaisen** [comment]
Would better add a issue and test case.

**Comment by EmmyMiao87** [comment]
#2483
#2493 
#2420

File: fe/src/main/java/org/apache/doris/analysis/StmtRewriter.java [hunk 1]
```diff
@@ -744,10 +744,17 @@ private static void canRewriteCorrelatedSubquery(
         Preconditions.checkNotNull(stmt);
         // Grouping and/or aggregation (including analytic functions) is only
         // allowed on correlated EXISTS subqueries
-        if ((expr instanceof BinaryPredicate
-                && (stmt.hasGroupByClause() || stmt.hasAnalyticInfo()))
-                || (expr instanceof InPredicate
-                && (stmt.hasAggInfo() || stmt.hasAnalyticInfo()))) {
+        if (expr instanceof BinaryPredicate) {
```
Comment (imay) [inline fe/src/main/java/org/apache/doris/analysis/StmtRewriter.java]: Should update corresponding comment.

File: fe/src/main/java/org/apache/doris/analysis/Expr.java [hunk 2]
```diff
@@ -123,6 +123,24 @@ public boolean apply(Expr arg) {
                 }
             };
     */
+    // Returns true if an Expr is a builtin aggregate function.
+    public final static com.google.common.base.Predicate<Expr> IS_BUILTIN_AGG_FN =
```
Comment (imay) [inline fe/src/main/java/org/apache/doris/analysis/Expr.java]: The name is a bit inaccurate, bitmap_union and hll_union are builtin aggregate function too.
So could you choose another name?

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**